src/AzureDevOps.DataIngestor.Sdk/Clients/VssGitClient.cs

```diff
@@ -20,23 +20,38 @@ internal VssGitClient(Uri baseUrl, VssCredentials credentials, ILogger logger) :
             this.retryAfter = VssClientHelper.GetRetryAfter(this.LastResponseContext);
         }
 
-        public async Task<List<GitCommitRef>> GetCommitsAsync(Guid repositoryId, string branchName, DateTime fromDate, DateTime toDate, int top = 100, int? skip = null)
+        public async Task<List<GitCommitRef>> GetCommitsWithRetryAsync(Guid repositoryId, string branchName, DateTime fromDate, DateTime toDate, int top = 100, int? skip = null)
         {
             GitQueryCommitsCriteria searchCriteria = new GitQueryCommitsCriteria
             {
                 FromDate = fromDate.ToUniversalTime().ToString("o"),
                 ToDate = toDate.ToUniversalTime().ToString("o"),
-                ItemVersion = new GitVersionDescriptor
+            };
+
+            if (!string.IsNullOrEmpty(branchName))
+            {
+                searchCriteria.ItemVersion = new GitVersionDescriptor
                 {
                     Version = branchName,
                     VersionType = GitVersionType.Branch
-                },
-            };
+                };
+            }
 
-            List<GitCommitRef> commitRefs = await RetryHelper.SleepAndRetry(this.retryAfter, this.logger, async () =>
-            {
-                return await this.GetCommitsAsync(repositoryId, searchCriteria, skip, top);
-            });
+            List<GitCommitRef> commitRefs = new List<GitCommitRef>();
+            try
+            {
+                commitRefs = await RetryHelper.SleepAndRetry(this.retryAfter, this.logger, async () =>
+                {
+                    return await this.GetCommitsAsync(repositoryId, searchCriteria, skip, top);
+                });

```

Comment:
If there are more than 100 commits within the date range, do we need to use a continuation token or retrieve more?

Reply:
For sure we need more, I missed the 100 default value - as off my screen somehow. I checked the API and don't see a continuationToken. Checking

Reply:
Done.